calcom/cal.com
# feat: configure availability time picker interval by env variable

## What does this PR do?
- Add an optional environment variable to changing the availability time options.    So starting times can be more precise than every 15 mintues.
- Falls back to default 15 minutes if no env set.
- Add the env variable `NEXT_PUBLIC_AVAILABILITY_SCHEDULE_INTERVAL`
  - Just followed the format of some of the other variables listed, feel free to change.

Fixes #12189
https://github.com/calcom/cal.com/issues/12189

## Issue
- Currently,  you can set an event to have a meeting length of any time.  In our case, 10 minutes.
- But availability time options in the Schedule can only be selected in 15 minute intervals. 
- In this scenario,  availability is desired to start to coincide with these 10 minute intervals.  
  - ie- bookings can start from 9.10am
  - setting a available time of 9am allows a booking time too early (ie-9am), and an availabile time of 9.15 misses the first  possible appointment of 9.10am.

## Type of change

## How should this be tested?
- set your env variable to the desired amount,  in our case we just wanted 10 minutes.
- A users availability options should now be able to be set using 10 minute intervals instead of 15.
With a booking
- setting an event-type with 10 minute intervals
- then setting a user with a starting time of 9.10 or any combination of start times that go off 10 mins instead of 15/

## Mandatory Tasks

## Checklist



- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
